proposals/cleanup.md

```diff
@@ -209,12 +209,52 @@ A couple implementations have been suggested for the annotation/label methodolog
+
+- We need to put the necessary information to perform the deletion in the work queue and make sure that the item should be visible in the queue when the delay calculated has elapsed.
+
+- Workers will then continuously pick up items from the queue and perform the actual object deletion.
+
+- We should use the single label only as there exists only AND based operator, so if we will be applying AND operator to select the resources based on the labels we need to include both `cleanup.kyverno.io/ttl` and `cleanup.kyverno.io/expires` as the labels in the resoource definition which is not we wanted, also there is no alternative to enable the OR based operator to select the resources to be deleted based on either of the two labels, and to enable OR based selection of resources based on the labels  we need to use the two informer factories and two set of controllers will be defined for each of the resources which will load a lot of burden on the system.
+
+- Also the absolute date format as defined in the above proposal is invalid and the default validation webhook of the cluster will throw an error so instead of `2022-08-04T00:30:00Z` as a label format, we should use `2022-08-04T003000Z` as a label format and we will handle the parsing of the label by defining a custom layout to parse this date.
+
+- Add the discovery api support which will automatically discover the resources available in the cluster and will also check that whether the resource are supporting get, list and delete verbs and also handled the skipping of the resources which do not support the listed verbs.
+
+- Used auth package to check whether the service account associated with resource has permissions to get, list and delete the particular resource.
+
+- Implement the logic for listening for the creation and deletion of the specific controllers when the particular resource is added or removed from the cluster and when there is the change in the RBAC permissions for the service account associated.
+
+- Implement the logic for graceful shutdown of the controller.
+
+- Made the custom informers so as to support the graceful shutdown of the individual controllers and also avoiding the resource leak.
+
+- Implmented the graceful shutdown of the workers by using the cancellable context.
+
+- Added the unit tests for the calculation and correect parsing of the time.
+
+- Added the Kuttl tests to make sure that each of the scenarios is covered gracefully by the controller.
+
+- Added the new validation-webhook to make sure that the resources with the label value follows the specific pattern, if not a warning is thrown to the user that the label to which the Kyverno is keeping a watch doesn't follow the recommneded format.
+
+- Created a webhook which will only keep a watch for the resources with the label `cleanup.kyverno.io/ttl` (used the labelSelector from the from the apimachinery/meta/v1 library to do so) and verfies that the label value adheres to the format defined or not.
+
+
```

Comment:
Please format this section better.
Things that are important, the label, the date format, the fact that we use only metadata, permissions checks, the manager that starts/stops individual controllers, etc...